src/vm_ppc.dasc

```diff
@@ -2586,39 +2762,72 @@ static void build_subroutines(BuildCtx *ctx)
   |//-----------------------------------------------------------------------
   |
   |.macro savex_, a, b, c, d
-  |  stfd f..a, 16+a*8(sp)
-  |  stfd f..b, 16+b*8(sp)
-  |  stfd f..c, 16+c*8(sp)
-  |  stfd f..d, 16+d*8(sp)
+  |  stfd f..a, EXIT_OFFSET+a*8(sp)
+  |  stfd f..b, EXIT_OFFSET+b*8(sp)
+  |  stfd f..c, EXIT_OFFSET+c*8(sp)
+  |  stfd f..d, EXIT_OFFSET+d*8(sp)
+  |.endmacro
+  |
+  |.macro saver, a
+  |  stp r..a, EXIT_OFFSET+32*8+a*PSIZE(sp)
   |.endmacro
   |
   |->vm_exit_handler:
   |.if JIT
-  |  addi sp, sp, -(16+32*8+32*4)
-  |  stmw r2, 16+32*8+2*4(sp)
+  |  addi sp, TMP0, sp, -(EXIT_OFFSET+32*8+32*PSIZE)
```

Comment:
FWIW, Debian is using the changes from this PR and the powerpc build [fails](https://buildd.debian.org/status/fetch.php?pkg=luajit&arch=powerpc&ver=2.1.0%7Ebeta3%2Bdfsg-5.1&stamp=1547954442&raw=0) because this `addi` instruction is incorrect:

```
make[4]: Entering directory '/<<PKGBUILDDIR>>/src'
HOSTCC    host/minilua.o
HOSTLINK  host/minilua
DYNASM    host/buildvm_arch.h
vm_ppc.dasc:2777: error: wrong number of parameters for `addi':
  |  addi sp, TMP0, sp, -(EXIT_OFFSET+32*8+32*PSIZE)
vm_ppc.dasc:*: info: 1 error in input file -- no output file generated.
make[4]: *** [Makefile:648: host/buildvm_arch.h] Error 1
```

Reply:
Ah, I missed that you already notified the author about this issue.